core/auth/src/main/java/software/amazon/awssdk/auth/credentials/AwsSessionCredentials.java

```diff
@@ -68,10 +78,20 @@ public String sessionToken() {
         return sessionToken;
     }
 
+    @Override
+    public Optional<String> accountId() {
+        return Optional.ofNullable(accountId);
+    }
+
+    public Optional<Instant> expiration() {
```

Comment:
If this makes sense to add to this interface, why not add it to AwsSessionCredentialsIdentity and `@Override` here.